libcodechecker/server/api/report_server.py

```diff
@@ -729,12 +728,24 @@ def getDiffResultsHash(self, run_ids, report_hashes, diff_type):
                         .except_(base_hashes).alias('new_bugs')
                     return [res[0] for res in session.query(new_hashes)]
                 else:
-                    new_hashes = union_all(*[
-                        select([bindparam('bug_id' + str(i), h)
-                               .label('bug_id')])
-                        for i, h in enumerate(report_hashes)])
-                    q = select([new_hashes]).except_(base_hashes)
-                    return [res[0] for res in session.query(q)]
+                    # The maximum number of compound select in sqlite is 500
+                    # by default. We increased SQLITE_MAX_COMPOUND_SELECT
+                    # limit but when the number of compound select was larger
+                    # than 8435 sqlite threw a `Segmentation fault` error.
+                    # For this reason we create queries with chunks.
+                    new_hashes = []
+                    chunk_size = 500
+                    for chunk in [report_hashes[i:i + chunk_size] for
+                                  i in range(0, len(report_hashes),
+                                             chunk_size)]:
+                        new_hashes_query = union_all(*[
+                            select([bindparam('bug_id' + str(i), h)
```

Comment:
why do you need bindparam(...) in this select expression?

Reply:
If you print the query out, you can see that this will build the following SQL query:
```sql
SELECT bug_id 
FROM (
  SELECT :bug_id0 AS bug_id
  UNION ALL
  SELECT :bug_id1 AS bug_id
) EXCEPT SELECT reports.bug_id AS bug_id 
FROM reports
LEFT OUTER JOIN files ON reports.file_id = files.id 
WHERE reports.run_id IN (:run_id_1)
```
`:bug_id0`, `:bug_id1` etc. are report hashes, which comes from the client. We should escape these values because the user can use it for SQL injection. That's the reason why I used `bindparam` here. 

For more information see the documentation: http://docs.sqlalchemy.org/en/latest/core/sqlelement.html#sqlalchemy.sql.expression.bindparam

Reply:
ah. OK. thx